ArrtModel/Model/ArrFrontend.cpp

```diff
@@ -42,60 +42,52 @@ void ArrFrontend::reconnectAccount()
 
 void ArrFrontend::connect()
 {
-    assert(m_reentryCnt.fetch_add(1) == 0 && m_sessionPropertiesAsync == nullptr);
+    assert(m_reentryCnt.fetch_add(1) == 0 && m_sessionPropertiesAsync != RR::Status::InProgress);
 
-    if (m_rrFrontend)
+    if (m_rrClient)
     {
         setStatus(AccountConnectionStatus::NotAuthenticated);
-        m_rrFrontend = nullptr;
+        m_rrClient = nullptr;
     }
 
     if (!m_region.empty() && !m_accountId.empty() && !m_accountKey.empty() && !m_accountDomain.empty())
     {
         setStatus(AccountConnectionStatus::CheckingCredentials);
 
-        RR::AzureFrontendAccountInfo fi{};
-        fi.AccountDomain = m_region;
+        RR::SessionConfiguration fi{};
+        fi.AccountDomain = m_accountDomain;
+        fi.RemoteRenderingDomain = m_region;
         fi.AccountId = m_accountId;
         fi.AccountKey = m_accountKey;
-        fi.AccountAuthenticationDomain = m_accountDomain;
 
-        auto frontend = RR::ApiHandle(RR::AzureFrontend(fi));
-        frontend->MessageLogged(&qArrSdkMessage);
-        frontend->SetLogLevel(RR::LogLevel::Debug);
+        auto client = RR::ApiHandle(RR::RemoteRenderingClient(fi));
+        client->MessageLogged(&qArrSdkMessage);
+        client->SetLogLevel(RR::LogLevel::Debug);
         QPointer<ArrFrontend> thisPtr = this;
-        auto async = frontend->GetCurrentRenderingSessionsAsync();
-        if (async)
-        {
-            m_sessionPropertiesAsync = async.value();
-            m_sessionPropertiesAsync->Completed([thisPtr, frontend](const RR::ApiHandle<RR::SessionPropertiesArrayAsync>& /*async*/) {
-                {
-                    std::unique_lock<std::mutex> lk(thisPtr->m_mutex);
-                    thisPtr->m_condVar.notify_one();
-                }
-                QMetaObject::invokeMethod(QApplication::instance(),
-                                          [thisPtr, frontend]() {
-                                              if (thisPtr != nullptr)
+        m_sessionPropertiesAsync = RR::Status::InProgress;
+        client->GetCurrentRenderingSessionsAsync([thisPtr, client](RR::Status status, RR::ApiHandle<RR::RenderingSessionPropertiesArrayResult> result) {
```

Comment:
Out of curiosity, what happens if there was a previous async call that is not completed yet? And is there a difference from the previous behaviour (was it cancelled if the async object was deleted?)

Reply:
The condition has not changed at all compared to previous version